src/H5CXprivate.h

```diff
@@ -47,6 +50,236 @@ typedef struct H5CX_state_t {
+    bool actual_selection_io_mode_valid; /* Whether actual selection I/O mode is valid */
+
+    /* Cached LCPL properties */
+    H5T_cset_t encoding;         /* Link name character encoding (H5P_STRCRT_CHAR_ENCODING_NAME) */
+    bool       encoding_valid;   /* Whether link name character encoding is valid */
+    unsigned intermediate_group; /* Whether to create intermediate groups (H5L_CRT_INTERMEDIATE_GROUP_NAME) */
+    bool     intermediate_group_valid; /* Whether create intermediate group flag is valid */
+
+    /* Cached LAPL properties */
+    size_t nlinks;       /* Number of soft / UD links to traverse (H5L_ACS_NLINKS_NAME) */
+    bool   nlinks_valid; /* Whether number of soft / UD links to traverse is valid */
+
+    /* Cached DCPL properties */
+    bool    do_min_dset_ohdr; /* Whether to minimize dataset object header (H5D_CRT_MIN_DSET_HDR_SIZE_NAME) */
+    bool    do_min_dset_ohdr_valid; /* Whether minimize dataset object header flag is valid */
+    uint8_t ohdr_flags;             /* Object header flags (H5O_CRT_OHDR_FLAGS_NAME) */
+    bool    ohdr_flags_valid;       /* Whether the object headers flags are valid */
+
+    /* Cached DAPL properties */
+    const char *extfile_prefix;       /* Prefix for external file (H5D_ACS_EFILE_PREFIX_NAME) */
+    bool        extfile_prefix_valid; /* Whether the prefix for external file is valid */
+    const char *vds_prefix;           /* Prefix for VDS (H5D_ACS_VDS_PREFIX_NAME) */
+    bool        vds_prefix_valid;     /* Whether the prefix for VDS is valid           */
+
+    /* Cached FAPL properties */
+    H5F_libver_t low_bound;       /* low_bound property for H5Pset_libver_bounds()
+                                     (H5F_ACS_LIBVER_LOW_BOUND_NAME) */
+    bool         low_bound_valid; /* Whether low_bound property is valid */
+    H5F_libver_t high_bound;      /* high_bound property for H5Pset_libver_bounds
+                                     (H5F_ACS_LIBVER_HIGH_BOUND_NAME) */
+    bool high_bound_valid;        /* Whether high_bound property is valid */
+
+    /* Cached VOL settings */
+    H5VL_connector_prop_t vol_connector_prop; /* Property for VOL connector ID & info
+                               This is treated as an independent field with
+                               no relation to the property H5F_ACS_VOL_CONN_NAME stored on the FAPL */
+    bool  vol_connector_prop_valid;           /* Whether property for VOL connector ID & info is valid */
+    void *vol_wrap_ctx;                       /* VOL connector's "wrap context" for creating IDs */
+    bool  vol_wrap_ctx_valid; /* Whether VOL connector's "wrap context" for creating IDs is valid */
+} H5CX_t;
+
+/* Typedef for nodes on the API context stack */
+/* Each entry into the library through an API routine invokes H5CX_push()
+ * in a FUNC_ENTER_API* macro, which pushes an H5CX_node_t on the API
+ * context [thread-local] stack, after initializing it with default values
+ * in H5CX__push_common().
```

Comment:
Comment references deleted routine `H5CX__push_common()`

Reply:
Updated in next push